pandas-stubs/core/algorithms.pyi

```diff
@@ -19,12 +21,21 @@ from pandas._typing import AnyArrayLike
 
 # These are type: ignored because the Index types overlap due to inheritance but indices
 # with extension types return the same type while standard type return ndarray
+
+_IntervalT = TypeVar(
+    "_IntervalT",
+    Interval[int],
+    Interval[float],
+    Interval[pd.Timestamp],
+    Interval[pd.Timedelta],
+)
+
 @overload
 def unique(values: PeriodIndex) -> PeriodIndex: ...  # type: ignore[misc]
 @overload
 def unique(values: CategoricalIndex) -> CategoricalIndex: ...  # type: ignore[misc]
 @overload
-def unique(values: IntervalIndex) -> IntervalIndex: ...  # type: ignore[misc]
+def unique(values: IntervalIndex[_IntervalT]) -> IntervalIndex[_IntervalT]: ...  # type: ignore[misc]
```

Comment:
error you're getting is because `pd.interval_range()` is returning `IntervalIndex`, so you could change the return type there, or just remove the generic declaration here (i.e., leave it as it was).

Reply:
I think this is fixed now.